aiida/backends/tests/cmdline/commands/test_code.py

```diff
@@ -36,23 +36,23 @@ def test_reachable(self):
 
     def test_interactive_remote(self):
         from aiida.orm import Code
-        os.environ['VISUAL'] = 'vim -cwq'
-        os.environ['EDITOR'] = 'vim -cwq'
+        os.environ['VISUAL'] = 'sleep 1; vim -cwq'
+        os.environ['EDITOR'] = 'sleep 1; vim -cwq'
```

Comment:
since time is of the essence for unit tests, what about `sleep 0.1`?

Reply:
Unfortunately the time precision on my Mac is 1 second, so a time >=1s is needed... Of course better approaches are accepted (even if probably they should become fixes to `click.edit()`

Reply:
I made a PR https://github.com/pallets/click/pull/1050 at click to fix this.
Once merged and released we can
- remove the `sleep 1`
- I would also substitute `vim` with `sed` (slimmer, no "window" open, ...) as I did in that PR

Reply:
vim was chosen because it is a file editor and therefore closer to the intended usage than the stream editor sed. Since this test turned out to be fragile, I agree that maybe sed is a better compromise between test reliability and testing the right thing.